### ISSUE TITLE
Suggestion: Node Text Object

### ISSUE THREAD

**Issue by chrisgrieser** [nvim-treesitter/nvim-treesitter-textobjects#333]
**Is your feature request related to a problem? Please describe.**
Rather generic text objects also seem to be useful, I use statement or block surprisingly often. I was thinking, that a simply text object "node" might be a nice addition to those.

**Describe the solution you'd like**
A node text object, which is the "smallest" node treesitter can recognize. For example, inner node would be a string without quotation marks, and outer node would be a node with quotation marks.

**Describe alternatives you've considered**
Creating a custom text object with a plugin like mini.ai, but that does not achieve the level of genericness required for this.

**Comment by theHamsta** [comment]
Something like this https://github.com/mfussenegger/nvim-treehopper or https://github.com/RRethy/nvim-treesitter-textsubjects?

This plugin should be able to do something similar with `(_) @small.outer _ @small.inner`

**Comment by chrisgrieser** [comment]
yeah, something in that regard, only text-subjects is "too small"
